public/app/features/scenes/components/VizPanel/VizPanelRenderer.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { RefCallback } from 'react';
+import { useMeasure } from 'react-use';
+
+import { PluginContextProvider } from '@grafana/data';
+import { PanelChrome, ErrorBoundaryAlert } from '@grafana/ui';
+import { appEvents } from 'app/core/core';
+import { useFieldOverrides } from 'app/features/panel/components/PanelRenderer';
+
+import { sceneGraph } from '../../core/sceneGraph';
+import { SceneComponentProps } from '../../core/types';
+import { SceneQueryRunner } from '../../querying/SceneQueryRunner';
+import { SceneDragHandle } from '../SceneDragHandle';
+
+import { VizPanel } from './VizPanel';
+
+export function VizPanelRenderer({ model }: SceneComponentProps<VizPanel>) {
+  const { title, options, fieldConfig, pluginId, pluginLoadError, $data, ...state } = model.useState();
+  const [ref, { width, height }] = useMeasure();
+  const plugin = model.getPlugin();
+  const { data } = sceneGraph.getData(model).useState();
+  const layout = sceneGraph.getLayout(model);
+
+  const isDraggable = layout.state.isDraggable ? state.isDraggable : false;
+  const dragHandle = <SceneDragHandle layoutKey={layout.state.key!} />;
+
+  const titleInterpolated = sceneGraph.interpolate(model, title);
+
+  // Not sure we need to subscribe to this state
+  const timeZone = sceneGraph.getTimeRange(model).state.timeZone;
+
+  const dataWithOverrides = useFieldOverrides(plugin, fieldConfig, data, timeZone);
+
+  if (pluginLoadError) {
+    return <div>Failed to load plugin: {pluginLoadError}</div>;
+  }
+
+  if (!plugin || !plugin.hasPluginId(pluginId)) {
+    return <div>Loading plugin panel...</div>;
+  }
+
+  if (!plugin.panel) {
+    return <div>Panel plugin has no panel component</div>;
+  }
+
+  const PanelComponent = plugin.panel;
+
+  // Query runner needs to with for auto maxDataPoints
+  if ($data instanceof SceneQueryRunner) {
+    $data.setContainerWidth(width);
+  }
+
+  return (
+    <div ref={ref as RefCallback<HTMLDivElement>} style={{ width: '100%', height: '100%' }}>
+      <PanelChrome
+        title={titleInterpolated}
+        width={width}
+        height={height}
+        leftItems={isDraggable ? [dragHandle] : undefined}
+      >
+        {(innerWidth, innerHeight) => (
+          <>
+            {!dataWithOverrides && <div>No data...</div>}
+            {dataWithOverrides && (
+              <ErrorBoundaryAlert dependencies={[plugin, data]}>
+                <PluginContextProvider meta={plugin.meta}>
+                  <PanelComponent
+                    id={1}
```

Comment:
Out of curiosity. Why are we fixing the `id` to a constant here?

Reply:
@ivanortegaalba it's a todo / temporary thing, the current VizPanel does not have a numerical id, we might need one for backward compatibility with the old dashboards but not added yet.